File: solutions/2026-04-21/solution.py
# First Missing Positive
# Difficulty: Medium
# URL: https://leetcode.com/problems/first-missing-positive/

class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        if not nums:
            return 1
        n = len(nums)
        nums = set(nums)
        for i in range(1, n + 2):
            if i not in nums:
                return i
        return n + 1

File: solutions/2026-04-21/solution.java
// First Missing Positive
// Difficulty: Medium
// URL: https://leetcode.com/problems/first-missing-positive/

class Solution {
    public int firstMissingPositive(int[] nums) {
        if (nums.length == 0) return 1;
        Set<Integer> set = new HashSet<>();
        for (int num : nums) set.add(num);
        for (int i = 1; i <= nums.length + 1; i++) {
            if (!set.contains(i)) return i;
        }
        return nums.length + 1;
    }
}